### PR TITLE
Change the trigger for the runtime/main -> runtimelab/runtime-main

### PR DESCRIPTION
**Pull request by pavelsavara**
try to trigger the flow from changes of internal/dotnet-runtime main branch instead of maestro

### CODE REVIEW

File: runtime-main-mirror.yml [hunk 1]
```diff
@@ -1,5 +1,16 @@
-# Disable triggers as it should only be called using Maestro
-trigger: none
+resources:
```
Comment (pavelsavara) [inline runtime-main-mirror.yml:1]: I'm trying to trigger it from https://dev.azure.com/dnceng/internal/_git/dotnet-runtime?path=/&version=GBmain&_a=history but I'm not sure about the syntax

File: runtime-main-mirror.yml [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+    type: git
+    name: internal/dotnet-runtime
+    ref: main
+    trigger:
+      branches:
+        include:
+        - main
+
+trigger:
+- main
```
Comment (mmitche) [inline runtime-main-mirror.yml:13]: I think this can be removed. You only need to trigger from runtime

File: runtime-main-mirror.yml (hunk 1 above)
Comment (mmitche) [reply runtime-main-mirror.yml:1]: I think this looks correct.

**Comment by mmitche** [comment]
@jkotas Who is responsible for runtimelab these days?

**Comment by jkotas** [comment]
> @jkotas Who is responsible for runtimelab these days?

If you are looking for who is responsible for fixing problems with runtimelab, it is generally the people who work on or sponsor currently active experimental project in dotnet/runtimelab. It includes myself, @pavelsavara, @kotlarmilos.

I think it would be fine to delete this synchronization job. It does not have a lot of value. If somebody wants to do merge from dotnet/runtime:main (it does not happen very often), it is only a few git commands to fetch the remote.

If you want to keep it, I think it should be triggered from public github only.

**Comment by pavelsavara** [comment]
> I think it would be fine to delete this synchronization job.

I like that idea.
~~That looks like nice way out of maintaining security token `BotAccount-dotnet-maestro-bot-PAT` used in https://dev.azure.com/dnceng/internal/_build?definitionId=906&_a=summary~~

There are other places where it's used.

> If you want to keep it, I think it should be triggered from public github only.

We have another code flow which does "fast forward" from github `runtime/main` to azdo `dotnet-runtime/main`. So I thought that triggering it of off azdo would be easier. 

As far as I can understand, triggering azdo directly from (another) github repo requires "service connection" and webhook on github side. See [docs](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/resources?view=azure-devops&tabs=schema#define-a-webhooks-resource)

I prefer not to do it this way.

File: runtime-main-mirror.yml (hunk 2 above)
Comment (pavelsavara) [reply runtime-main-mirror.yml:13]: this is pointing to the `ref: main` in the `resource`. I think that just adding resource is not enough.

**Pull request closed without merging**